### PR TITLE
Add test plan for sycl::kernel

### PR DESCRIPTION
**Pull request by vladimirkhashev**
Most of sycl::kernel functionality covered by tests https://github.com/KhronosGroup/SYCL-CTS/tree/SYCL-2020/tests/kernel.
This is test plan for remaining gaps.

### CODE REVIEW

File: test_plans/kernel.asciidoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+== Tests
+
+=== Deleted default constructor
+
+Check that std::is_default_constructible_v<sycl::kernel> is `false`.
+
+=== Test for `get_backend()`
```
Comment (gmlueck) [inline test_plans/kernel.asciidoc:23]: In all of the following tests it's not clear how you will get the `kernel` object.  You could add some common description that applies to all of the tests like:

> * Define a simple kernel named `KernelName` that uses no optional features.
> * Obtain a kernel bundle in `executable` state that contains this kernel via `get_kernel_bundle<KernelName, bundle_state::executabe(const context & const device &)`.
> * Get a `kernel` object for the kernel by calling `kernel_bundle::get_kernel<KernelName>()`.

Comment (vladimirkhashev) [reply test_plans/kernel.asciidoc:30]: Thank you, clarification how to obtain object is added.

**Review by gmlueck — APPROVED** [review]

**Comment by fraggamuffin** [comment]
LGTM

**Comment by fraggamuffin** [comment]
LGTM

File: test_plans/kernel.asciidoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+== Tests
+
+To get kernel object:
+
+* Define a simple kernel named `KernelName` that uses no optional features.
+* Obtain a kernel bundle in executable state that contains this kernel via
+`get_kernel_bundle<KernelName, bundle_state::executabe(const context & const device &)`.
```
Comment (keryell) [inline test_plans/kernel.asciidoc:23]:
```suggestion
`get_kernel_bundle<KernelName, bundle_state::executable(const context&, const device&)`.
```
Try to use some copy/paste from the specification. It is safer. :-)

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**